stestr/commands/run.py

```diff
@@ -177,6 +187,8 @@ def take_action(self, parsed_args):
             suppress_attachments = args.suppress_attachments
         verbose_level = self.app.options.verbose_level
         stdout = open(os.devnull, 'w') if verbose_level == 0 else sys.stdout
+        # Make sure all (python) callers have provided an int()
+        concurrency = to_int(concurrency)
```

Comment:
I think we don't need this here but in `run_command()` after
https://github.com/mtreinish/stestr/pull/232/files#diff-e77977ac6a427e77fc880ed584215c86R333

Because it must be `int` value with this.
https://github.com/mtreinish/stestr/pull/232/files#diff-e77977ac6a427e77fc880ed584215c86R62

Reply:
It needs to be here or in the case args.concurrency = '1' we'll set concurrency to '1' and then we'll compare '1' < 0 on line 192 and then python will complain that we're comparing a str() with an int()

Reply:
> It needs to be here or in the case args.concurrency = '1' we'll set concurrency to '1' and then we'll compare '1' < 0 on line 192 and then python will complain that we're comparing a str() with an int()

Yeah, but this should be there(on line 333) at least. Otherwise, we can't handle it like this.

>     File "/home/masayuki/git/stestr/stestr/commands/run.py", line 333, in run_command
>     if concurrency and concurrency < 0:

We actually expose `run_command()` function and tempest use it directly.

```
$ tempest --debug run  --regex '\[.*\bsmoke\b.*\]' --concurrency=4
'<' not supported between instances of 'str' and 'int'
Traceback (most recent call last):
  File "/home/masayuki/git/openstack/tempest/.tox/py37/lib/python3.7/site-packages/cliff/app.py", line 401, in run_subcommand
    result = cmd.run(parsed_args)
  File "/home/masayuki/git/openstack/tempest/.tox/py37/lib/python3.7/site-packages/cliff/command.py", line 184, in run
    return_code = self.take_action(parsed_args) or 0
  File "/home/masayuki/git/openstack/tempest/tempest/cmd/run.py", line 180, in take_action
    load_list=parsed_args.load_list, combine=parsed_args.combine)
  File "/home/masayuki/git/stestr/stestr/commands/run.py", line 333, in run_command
    if concurrency and concurrency < 0:
TypeError: '<' not supported between instances of 'str' and 'int'
Traceback (most recent call last):
  File "/home/masayuki/git/openstack/tempest/.tox/py37/bin/tempest", line 10, in <module>
    sys.exit(main())
  File "/home/masayuki/git/openstack/tempest/tempest/cmd/main.py", line 48, in main
    return the_app.run(argv)
  File "/home/masayuki/git/openstack/tempest/.tox/py37/lib/python3.7/site-packages/cliff/app.py", line 281, in run
    result = self.run_subcommand(remainder)
  File "/home/masayuki/git/openstack/tempest/.tox/py37/lib/python3.7/site-packages/cliff/app.py", line 401, in run_subcommand
    result = cmd.run(parsed_args)
  File "/home/masayuki/git/openstack/tempest/.tox/py37/lib/python3.7/site-packages/cliff/command.py", line 184, in run
    return_code = self.take_action(parsed_args) or 0
  File "/home/masayuki/git/openstack/tempest/tempest/cmd/run.py", line 180, in take_action
    load_list=parsed_args.load_list, combine=parsed_args.combine)
  File "/home/masayuki/git/stestr/stestr/commands/run.py", line 333, in run_command
    if concurrency and concurrency < 0:
TypeError: '<' not supported between instances of 'str' and 'int'
```

Reply:
Ahh okay I see.  I need it in both places.  I'll re-spin tomorrow if not beaten to it ;P

Reply:
> Ahh okay I see. I need it in both places. I'll re-spin tomorrow if not beaten to it ;P

heh. But well, I think we don't need it here because the value type must be int.

```
        parser.add_argument("--concurrency", action="store", default=None,
                            type=int,
                            help="How many processes to use. The default (0) "
                            "autodetects your CPU count.")
```

Reply:
I've updated the PR to do the checking in both places for the CLI and the Python API so we can get 2.3.1 out the door today.

@masayukig I think you're right Cliff/argparse will do the type enforcement for us, but I left it in place because I think the function will give us a good place to handle #202 when we decide on what the grammar should be for that.